cmd/web.go

```diff
@@ -170,23 +169,23 @@ func runWeb(ctx *cli.Context) error {
 	var err error
 	switch setting.Protocol {
 	case setting.HTTP:
-		err = runHTTP(listenAddr, context2.ClearHandler(m))
+		err = runHTTP(listenAddr, m)
 	case setting.HTTPS:
 		if setting.EnableLetsEncrypt {
-			err = runLetsEncrypt(listenAddr, setting.Domain, setting.LetsEncryptDirectory, setting.LetsEncryptEmail, context2.ClearHandler(m))
+			err = runLetsEncrypt(listenAddr, setting.Domain, setting.LetsEncryptDirectory, setting.LetsEncryptEmail, m)
 			break
 		}
 		if setting.RedirectOtherPort {
 			go runHTTPRedirector()
 		}
-		err = runHTTPS(listenAddr, setting.CertFile, setting.KeyFile, context2.ClearHandler(m))
+		err = runHTTPS(listenAddr, setting.CertFile, setting.KeyFile, m)
 	case setting.FCGI:
 		listener, err := net.Listen("tcp", listenAddr)
 		if err != nil {
 			log.Fatal(4, "Failed to bind %s", listenAddr, err)
 		}
 		defer listener.Close()
-		err = fcgi.Serve(listener, context2.ClearHandler(m))
```

Comment:
`context2.ClearHandler()`  is used for to prevent memory leaks. I think it may still be needed.